src/gymmick_init.c

```diff
@@ -0,0 +1,132 @@
+#include "global.h"
+#include "field_system.h"
+#include "gymmick.h"
+#include "gymmick_init.h"
+#include "math_util.h"
+#include "player_data.h"
+#include "save.h"
+#include "save_misc_data.h"
+#include "constants/badge.h"
+
+#define NUM_TRASH_CANS 15
+
+typedef struct TrashCan {
+    int numNeighbors;
+    u8 neighbors[4];
+} TrashCan;
+
+static const TrashCan sTrashCans[NUM_TRASH_CANS] = {
+    [0]  = {.numNeighbors = 2, .neighbors = { 1,   5,  0xFF, 0xFF }},
+    [1]  = {.numNeighbors = 3, .neighbors = { 0,   2,     6, 0xFF }},
+    [2]  = {.numNeighbors = 3, .neighbors = { 1,   3,     7, 0xFF }},
+    [3]  = {.numNeighbors = 3, .neighbors = { 2,   4,     8, 0xFF }},
+    [4]  = {.numNeighbors = 2, .neighbors = { 3,   9,  0xFF, 0xFF }},
+    [5]  = {.numNeighbors = 3, .neighbors = { 0,   6,    10, 0xFF }},
+    [6]  = {.numNeighbors = 4, .neighbors = { 1,   5,     7,   11 }},
+    [7]  = {.numNeighbors = 4, .neighbors = { 2,   6,     8,   12 }},
+    [8]  = {.numNeighbors = 4, .neighbors = { 3,   7,     9,   13 }},
+    [9]  = {.numNeighbors = 3, .neighbors = { 4,   8,    14, 0xFF }},
+    [10] = {.numNeighbors = 2, .neighbors = { 5,   11, 0xFF, 0xFF }},
+    [11] = {.numNeighbors = 3, .neighbors = { 6,   10,   12, 0xFF }},
+    [12] = {.numNeighbors = 3, .neighbors = { 7,   11,   13, 0xFF }},
+    [13] = {.numNeighbors = 3, .neighbors = { 8,   12,   14, 0xFF }},
+    [14] = {.numNeighbors = 2, .neighbors = { 9 ,  13, 0xFF, 0xFF }},
+};
+
+void InitEcruteakGym(FieldSystem *fieldSystem) {
+    struct Gymmick *gymmick = Save_GetGymmickPtr(FieldSystem_GetSaveData(fieldSystem));
+    Save_Gymmick_Init(gymmick, GYMMICK_ECRUTEAK);
+    
+    union GymmickUnion *data = Save_Gymmick_AssertMagic_GetData(gymmick, GYMMICK_ECRUTEAK);
+    for (int i = 0; i < 4; i++) {
+        data->ecruteak.candles[i] = 0;
+    }
+}
+
+void InitCianwoodGym(FieldSystem *fieldSystem) {
+    struct Gymmick *gymmick = Save_GetGymmickPtr(FieldSystem_GetSaveData(fieldSystem));
```

Comment:
Inconsistent use of `struct` type (ie why use struct for Gymmick and not FieldSystem). Please remove for this and all other instances